lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -644,4 +670,153 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
+                                    StatusEventType.PUBLISH_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableisPublic(latestIsPublic).build();
+                } else {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                    StatusEventType.UNPUBLISH_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).
+                            withNullableVersion(ev.getVersion().get()).
+                            withNullableisPublic(latestIsPublic).build();
+                }
+
+            } catch (IOException ex) {
+                throw handleException(ex);
+            } catch (JsonClientException ex) {
+                throw handleException(ex);
+            }
+        }
+        // update a (UN)DELETE_ALL_VERSIONS event
+        if (eventType.equals(StatusEventType.DELETE_ALL_VERSIONS) ||
+                eventType.equals(StatusEventType.UNDELETE_ALL_VERSIONS )) {
+            Map<String, Object> command;
+
+            // if source data object does not exist then update event as a
+            // DELETE_ALL_VERSIONS event
+            command = new HashMap<>();
+            command.put("command", "listObjects");
+            command.put("params", new ListObjectsParams().
+                    withIds(Arrays.asList(Long.decode(accessGrpId))));
+            try {
+                boolean objidExists = false;
+
+                final List<List> objList = ws.getClient().administer(new UObject(command))
+                        .asClassInstance(List.class);
+
+                for (List obj: objList) {
+                    long id = (long)((Integer)obj.get(0)).intValue();
+                    if (id  == Long.decode(objectId)) {
+                        objidExists = true;
+                        break;
+                    }
+                }
+
+                if (!objidExists) {
+                    return StatusEvent.
+                            getBuilder(ev.getStorageObjectType().get(), ev.getTimestamp(),
+                                    StatusEventType.DELETE_ALL_VERSIONS).
+                            withNullableAccessGroupID(Integer.decode(accessGrpId)).
+                            withNullableObjectID(objectId).build();
+                }
+            } catch (IOException ex) {
+                throw handleException(ex);
+            } catch (JsonClientException ex) {
+                throw handleException(ex);
+            }
+
+            // if source data object is deleted then update event as a DELETE_ALL_VERSIONS event
+            command = new HashMap<>();
```

Comment:
Same thing re max_obj_id and min_obj_id params t

Reply:
Done.